Tests/GrimoireKitTests/FileSpec.swift

```diff
@@ -0,0 +1,33 @@
+import Quick
+import Nimble
+import Foundation
+@testable import GrimoireKit
+
+final class FileSpec: QuickSpec {
+  override func spec() {
+    describe("read") {
+      context("when file path is faulty") {
+        let path = "fo  oo / bba r / bazz"
+
+        it("throws the correct error") {
+          expect { try File.read(path) }.to(throwError { error in
+            if case let File.Error.invalidFilePath(actual) = error {
+              expect(actual).to(equal("file://" + path))
```

Comment:
`expect(actual) == "file://\(path)"` ?

Reply:
Should we implement `Equatable` for `File.Error` ? Would make the whole test clearer.